ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/Utils.java

```diff
@@ -189,7 +189,8 @@ public static boolean shouldReplicate(ReplicationSpec replicationSpec, Table tab
     }
 
     // if its metadata only, then dump metadata of non native tables also.
-    if (tableHandle.isNonNative() && !replicationSpec.isMetadataOnly()) {
+    if (tableHandle.isNonNative() && !replicationSpec.isMetadataOnly()
+            && !Utils.shouldDumpMetaDataOnlyForExternalTables(tableHandle, hiveConf)) {
```

Comment:
The check is very specific to DAS ..it includes external table meta data even if external table replication is not enabled ..i am not sure if we can have same for this config also